uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -379,9 +397,37 @@ define([
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::updateBalance', function () {
-          vm.updateBalance();
-          setDaySelectionMode();
+        $rootScope.$on('LeaveRequestPopup::updateAbsenceType', function () {
+          var absenceTypeUnitChanged;
+          var previousAbsenceTypeUnit = vm.selectedAbsenceType.calculation_unit_name;
+
+          vm.selectedAbsenceType = getSelectedAbsenceType();
+          absenceTypeUnitChanged = previousAbsenceTypeUnit !== vm.selectedAbsenceType.calculation_unit_name;
+
+          setOpeningBalance();
+
+          return $q.resolve()
+            .then(absenceTypeUnitChanged && setDaysSelectionMode)
+            .then(function () {
+              if (absenceTypeUnitChanged && isCalculationUnit('hours')) {
+                return loadTimeRangesFromWorkPattern('from');
+              }
+            })
+            .then(setRequestDates)
+            .then(performBalanceChangeCalculation);
+        }),
+        $rootScope.$on('LeaveRequestPopup::updateBalance', function (event, absenceTypesWithBalances) {
+          vm.absenceTypes = absenceTypesWithBalances;
+          vm.selectedAbsenceType = getSelectedAbsenceType();
+
+          if (moment(vm.uiOptions.toDate).isAfter(vm.period.end_date)) {
```

Comment:
again, this is all logic related to dates, what does it have to do with updating the balance? i begin to suspect that the name of this event is wrong

Reply:
@AkA84 could it be not `::updateBalance` but `::updateAbsencePeriod`?

Reply:
Renamed to `::updateAbsencePeriodAndBalances`